src/goto-symex/symex_target_equation.cpp

```diff
@@ -319,27 +321,52 @@ void symex_target_equationt::constraint(
 }
 
 void symex_target_equationt::convert_without_assertions(
-  decision_proceduret &decision_procedure)
+  decision_proceduret &decision_procedure,
+  solver_hardnesst *solver_hardness)
+{
+  if(solver_hardness)
+    solver_hardness->register_ssa_size(SSA_steps.size());
+
+  convert_guards(decision_procedure, solver_hardness);
+  convert_assignments(decision_procedure, solver_hardness);
+  convert_decls(decision_procedure, solver_hardness);
+  convert_assumptions(decision_procedure, solver_hardness);
+  convert_goto_instructions(decision_procedure, solver_hardness);
+  convert_function_calls(decision_procedure, solver_hardness);
+  convert_io(decision_procedure, solver_hardness);
+  convert_constraints(decision_procedure, solver_hardness);
+}
+
+void symex_target_equationt::convert(
+  decision_proceduret &decision_procedure,
+  propt &prop)
 {
-  convert_guards(decision_procedure);
-  convert_assignments(decision_procedure);
-  convert_decls(decision_procedure);
-  convert_assumptions(decision_procedure);
-  convert_goto_instructions(decision_procedure);
-  convert_function_calls(decision_procedure);
-  convert_io(decision_procedure);
-  convert_constraints(decision_procedure);
+  auto solver_hardness = [&prop]() -> solver_hardnesst * {
+    if(auto hardness_collector = dynamic_cast<hardness_collectort *>(&prop))
```

Comment:
🚫 Make `prop_conv_solvert` implement `hardness_collectort` and forward the calls. `propt` MUST not be exposed here.